### ISSUE TITLE
Add Fjord Launcher to the Scoop installer

### ISSUE THREAD

**Issue by TechPro424** [unmojang/FjordLauncher#2]
### Role

I play cracked modded Minecraft.

### Suggestion

I want Fjord Launcher to be available as a Scoop (https://scoop.sh/) app

### Benefit

So that I can install/update it in the CLI

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

I could add it to my Scoop bucket if you're fine with it.
If yes, I have some questions:

- Does Fjord Launcher use the prismlauncher appdata directory or its own?
- If it uses its own directory, what is the exact name of that directory?
- Also, does it use the prismlauncher.cfg file or its own?
- If it uses its own .cfg file, what is the exact name of that file?

**Comment by evan-goode** [comment]
I made a Scoop bucket here: https://github.com/unmojang/scoop-unmojang; I'm going to package Fjord for a few more systems and then add them all to an "Installation" section in the README.

> I could add it to my Scoop bucket if you're fine with it. If yes, I have some questions:

Of course, you're more than welcome to. And/or help maintain unmojang/scoop-umojang :) I would definitely appreciate the help.

> 
>     * Does Fjord Launcher use the prismlauncher appdata directory or its own?
> 
>     * If it uses its own directory, what is the exact name of that directory?

Its own: `C:\Users\user\AppData\Roaming\FjordLauncher`. When installed with Scoop, it uses `C:\Users\user\scoop\persist\fjordlauncher`, similar to Prism Launcher.

> 
>     * Also, does it use the prismlauncher.cfg file or its own?
> 
>     * If it uses its own .cfg file, what is the exact name of that file?

It uses `C:\Users\user\AppData\Roaming\FjordLauncher\fjordlauncher.cfg`.

To make the Scoop package, I just did a find/replace on the upstream Prism Launcher package to replace "Prism" with "Fjord" (and other case variants, too). Going forward, I see no reason the package might stray further from upstream than that.

Also, unlike PollyMC, Fjord can auto-migrate the data directory from Prism Launcher (I think PollyMC only supported migration from PolyMC and MultiMC). But I'm not sure that works when using Scoop.

**Comment by TechPro424** [comment]
Great! Let me know if you need any help with the Scoop bucket, I'll be happy to assist.
Also FYI, installing PollyMC from my Scoop bucket should migrate the data from Prism

Please add the Scoop install details to the README

And also while I'm here, does Fjord support Microsoft accounts too or only offline mode?

**Comment by evan-goode** [comment]
> Please add the Scoop install details to the README

I'll keep this issue open until I do.

> And also while I'm here, does Fjord support Microsoft accounts too or only offline mode?

It supports Microsoft accounts the same as PollyMC, you have to supply your own MSA client ID. In the future, I would like to switch to using the client ID from Mojang's official launcher so the user doesn't have to bring their own.

**Comment by TechPro424** [comment]
> supply your own MSA client ID.

What does that mean, and how do I go about doing it?

**Comment by evan-goode** [comment]
Go to Settings -> APIs -> API Keys and enter a client ID under "Microsoft Authentication". For example, PolyMC's client ID can be found here: https://github.com/PolyMC/PolyMC/blob/62fd3d436a204232a5ae3bf9be6fde5fec819b81/CMakeLists.txt#L133.

**Comment by TechPro424** [comment]
Is there a way to get my own?

**Comment by evan-goode** [comment]
Yes, though I'm not exactly sure what the process is like. Try starting here? https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app

**Comment by TechPro424** [comment]
Is there a chance of Fjord getting a client ID?

**Comment by evan-goode** [comment]
> Is there a chance of Fjord getting a client ID?

Possibly, we could get our own, but I would rather come up with a way to fetch one at runtime from Mojang's own launcher files, that way we can avoid having an awkward proprietary-ish 36-byte blob in the codebase that can't be used in forks. I've started work on that, but it'll take some time.

Closing this issue since I've added Scoop install instructions to the README: https://github.com/unmojang/FjordLauncher#scoop-recommended.
